src/components/PreviewPane/PreviewPane.js

```diff
@@ -50,6 +51,35 @@ export default class PreviewPane extends React.Component {
     });
   };
 
+  widgetsFor = (name) => {
+    const { fields, entry, getAsset } = this.props;
+    const field = fields && fields.find(f => f.get('name') === name);
+    const nestedFields = field && field.get('fields');
+    const value = entry.getIn(['data', field.get('name')]);
+
+    const widgetFor = (field, value) => {
```

Comment:
can this function be extracted from the main `widgetsFor`? We should try to avoid inline functions as much as possible.